### PR TITLE
CMake support updated for the latest version

### PR DESCRIPTION
**Pull request by maximmenshikov**
That's the latest effort to bring CMake to RuC compilation process. This time it supports all submodules (compiler, VM, UI, util), and doesn't break the build and/or testing.

Legacy Makefile wrapping CMake is provided, one can use "make" to build the project.
Travis testing doesn't use cmake test because this is questionable, so we fallback to previous method.

As for FLTK, you can add -DFLTK_IMPORTED=OFF to use system-wide library. That might dramatically speed up compilation.

This version is taking some parts from @broadwaylamb (thank you), however, a lot of parts are remade.

### CODE REVIEW

**Comment by maximmenshikov** [comment]
Minor bug in CI, working on a fix

**Comment by maximmenshikov** [comment]
The issue is fixed. Additionally I've added Sublime Text project file which adds a proper autocomplete based on CMake-generated compile_commands.json.

File: test.sh [hunk 1]
```diff
@@ -39,25 +46,24 @@ do
             if ! [[ -z $full_out ]] ; then
                 sleep $output_time
                 echo -e "\e[1;31m build failing \e[1;39m: $code"
```
Comment (MikhailTerekhov) [inline test.sh]: Console output coloring somewhy doesn't work in macOS. We might either fix it or turn it off there, such that '\e[1;32m' wouldn't show up in Travis output.

**Comment by MikhailTerekhov** [comment]
Thank you! Now it is almost ready, I will merge it as soon as Travis colored output issue is fixed.

> As for FLTK, you can add -DFLTK_IMPORTED=OFF to use system-wide library. That might dramatically speed up compilation.

Looks like this could be made the default behavior. We might even remove the FLTK submodule entirely, if system-wide FLTK could be easily installed on macOS too (haven't checked that yet).

File: test.sh (hunk 1 above)
Comment (maximmenshikov) [reply test.sh]: Not that it is related to cmake changeset, but I fixed it. See newest travis output.

**Comment by maximmenshikov** [comment]
> Thank you! Now it is almost ready, I will merge it as soon as Travis colored output issue is fixed.
> 
> > As for FLTK, you can add -DFLTK_IMPORTED=OFF to use system-wide library. That might dramatically speed up compilation.
> 
> Looks like this could be made the default behavior. We might even remove the FLTK submodule entirely, if system-wide FLTK could be easily installed on macOS too (haven't checked that yet).

Please apply it via a separate patch once you check it. You can start with switching default value from ON to OFF in main CMakeLists.
The advantage of this patch is that it is compatible with existing environments, being seamless drop-in update.

**Pull request merged**